app/renderer/components/navigation/navigator.js

```diff
@@ -0,0 +1,324 @@
+        }
+        <div className='navigatorWrapper'
+          onDoubleClick={this.onDoubleClick}
+          onDragOver={this.onDragOver}
+          onDrop={this.onDrop}
+        >
+          <div className='backforward'>
+            <div className={cx({
+              navigationButtonContainer: true,
+              nav: true,
+              disabled: !this.props.activeTab || !this.props.activeTab.get('canGoBack') || activeTabShowingMessageBox
+            })}>
+              <LongPressButton
+                l10nId='backButton'
+                className='navigationButton backButton'
+                disabled={!this.props.activeTab || !this.props.activeTab.get('canGoBack') || activeTabShowingMessageBox}
+                onClick={this.onBack}
+                onLongPress={this.onBackLongPress}
+              />
+            </div>
+            <div className={cx({
+              navigationButtonContainer: true,
+              nav: true,
+              disabled: !this.props.activeTab || !this.props.activeTab.get('canGoForward') || activeTabShowingMessageBox
+            })}>
+              <LongPressButton
+                l10nId='forwardButton'
+                className='navigationButton forwardButton'
+                disabled={!this.props.activeTab || !this.props.activeTab.get('canGoForward') || activeTabShowingMessageBox}
+                onClick={this.onForward}
+                onLongPress={this.onForwardLongPress}
+              />
+            </div>
+          </div>
+          <NavigationBar
+            sites={this.props.appState.get('sites')}
+            scriptsBlocked={activeFrame && activeFrame.getIn(['noScript', 'blocked'])}
+            partitionNumber={(activeFrame && activeFrame.get('partitionNumber')) || 0}
+            enableNoScript={siteUtil.enableNoScript(this.props.activeSiteSettings, this.props.appState)}
+            settings={this.props.appState.get('settings')}
+            noScriptIsVisible={noScriptIsVisible}
+            menubarVisible={this.props.customTitlebar.menubarVisible}
+            siteSettings={this.props.appState.get('siteSettings')}
+            synopsis={this.props.appState.getIn(['publisherInfo', 'synopsis']) || new Immutable.Map()}
+            locationInfo={this.props.appState.get('locationInfo')}
+          />
+          <div className='topLevelEndButtons'>
+            <div className={cx({
+              extraDragArea: !this.props.customTitlebar.menubarVisible,
+              allowDragging: this.props.shouldAllowWindowDrag
+            })} />
+            {
+              activeTabShowingMessageBox
+                ? null
+                : this.extensionButtons
+            }
+            <div className={css(styles.braveMenuContainer)}>
+              <Button iconClass='braveMenu'
+                l10nId='braveMenu'
+                testId='braveShieldButton'
+                className={cx({
+                  navbutton: true,
+                  braveShieldsDisabled: shieldDisabled,
```

Comment:
I think `braveShieldsDisabled` and `braveShieldsDown` can be too easily confused and the button doesn't need to know or care why shields are disabled. I would suggest combining into a single setting and method that handles both cases